library/pkparse.c

```diff
@@ -446,6 +446,154 @@ static int pk_group_id_from_specified(const mbedtls_asn1_buf *params,
 }
 #endif /* MBEDTLS_PK_PARSE_EC_EXTENDED */
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+/*
+ * Use EC parameters to initialise an EC group
+ *
+ * ECParameters ::= CHOICE {
+ *   namedCurve         OBJECT IDENTIFIER
+ *   specifiedCurve     SpecifiedECDomain -- = SEQUENCE { ... }
+ *   -- implicitCurve   NULL
+ */
+static int pk_use_ecparams(const mbedtls_asn1_buf *params, mbedtls_pk_context *pk)
+{
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    mbedtls_ecp_group_id grp_id;
+    psa_ecc_family_t ec_family;
+    size_t bits;
+
+    if (params->tag == MBEDTLS_ASN1_OID) {
+        if (mbedtls_oid_get_ec_grp(params, &grp_id) != 0) {
+            return MBEDTLS_ERR_PK_UNKNOWN_NAMED_CURVE;
+        }
+    } else {
+#if defined(MBEDTLS_PK_PARSE_EC_EXTENDED)
+        if ((ret = pk_group_id_from_specified(params, &grp_id)) != 0) {
+            return ret;
+        }
+#else
+        return MBEDTLS_ERR_PK_KEY_INVALID_FORMAT;
+#endif
```

Comment:
As in other places comments are added. I suggest to add also here.
```suggestion
#endif /* MBEDTLS_PK_PARSE_EC_EXTENDED */
```